### PR TITLE
[SPARK-6488] [MLlib] [Python] Support addition/multiplication in PySpark's BlockMatrix

### PR DESCRIPTION
**Pull request by dusenberrymw**
This PR adds addition and multiplication to PySpark's `BlockMatrix` class via `add` and `multiply` functions.

### CODE REVIEW

File: python/pyspark/mllib/linalg/distributed.py [hunk 1]
```diff
@@ -775,6 +775,80 @@ def numCols(self):
...
+        """
+        Adds two block matrices together. The matrices must have the
+        same size and matching `rowsPerBlock` and `colsPerBlock` values.
+        If one of the sub matrix blocks that are being added is a
+        SparseMatrix, the resulting sub matrix block will also be a
+        SparseMatrix, even if it is being added to a DenseMatrix. If
+        two dense sub matrix blocks are added, the output block will
+        also be a DenseMatrix.
+
+        :param other: A BlockMatrix
```
Comment (mengxr) [inline python/pyspark/mllib/linalg/distributed.py]: `param other` and `return` is not necessary here. `A BlockMatrix` doesn't provide much info.

**Comment by dusenberrymw** [comment]
@mengxr Do you have any additional thoughts on this, or is it ready to be merged?

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Comment by dusenberrymw** [comment]
Great, thanks!

**Pull request closed without merging**